GoogleDataTransport/GDTCORLibrary/GDTCOREvent.m

```diff
@@ -24,60 +24,32 @@
 
 #import "GDTCORLibrary/Private/GDTCOREvent_Private.h"
 
-static NSString *const kNextEventIDKey = @"GDTCOREventEventIDCounter";
-
 @implementation GDTCOREvent
 
-+ (void)nextEventIDForTarget:(GDTCORTarget)target
-                  onComplete:(void (^)(NSNumber *_Nonnull eventID))onComplete {
-  __block int32_t lastEventID = -1;
-  id<GDTCORStorageProtocol> storage = GDTCORStorageInstanceForTarget(target);
-  [storage libraryDataForKey:kNextEventIDKey
-      onFetchComplete:^(NSData *_Nullable data, NSError *_Nullable getValueError) {
-        if (getValueError != nil || data == nil || data.length == 0) {
-          lastEventID = 1;
-        } else {
-          [data getBytes:(void *)&lastEventID length:sizeof(int32_t)];
-        }
-        if (onComplete) {
-          onComplete(@(lastEventID));
-        }
-      }
-      setNewValue:^NSData *_Nullable(void) {
-        if (lastEventID != -1) {
-          int32_t incrementedValue = lastEventID + 1;
-          return [NSData dataWithBytes:&incrementedValue length:sizeof(int32_t)];
-        }
-        return nil;
-      }];
++ (NSString *)nextEventID {
+  // TODO: Consider a way to make the eventIDs incremental without introducing a storage dependency
+  // to the object.
+  return [NSUUID UUID].UUIDString;
 }
 
 - (nullable instancetype)initWithMappingID:(NSString *)mappingID target:(GDTCORTarget)target {
   GDTCORAssert(mappingID.length > 0, @"Please give a valid mapping ID");
   GDTCORAssert(target > 0, @"A target cannot be negative or 0");
-  __block NSNumber *eventID;
-  dispatch_semaphore_t sema = dispatch_semaphore_create(0);
-  [GDTCOREvent nextEventIDForTarget:target
-                         onComplete:^(NSNumber *_Nullable newEventID) {
-                           eventID = newEventID;
-                           dispatch_semaphore_signal(sema);
-                         }];
-  if (dispatch_semaphore_wait(sema, dispatch_time(DISPATCH_TIME_NOW, 5 * NSEC_PER_SEC)) != 0) {
-    return nil;
-  }
-  if (mappingID == nil || mappingID.length == 0 || target <= 0 || eventID == nil) {
+  if (mappingID == nil || mappingID.length == 0 || target <= 0) {
```

Comment:
wouldn't mappingID.length > 0 accomplish the first two check?

Reply:
Yeah, good idea, I just removed `eventID == nil`, but I'll update this part as well.

Reply:
Done